src/sql/workbench/browser/modal/calloutDialog.ts

```diff
@@ -103,224 +52,32 @@ export class CalloutDialog extends Modal {
-			let selectedUri = await this.handleBrowse();
-			if (selectedUri) {
-				this._imageUrlInputBox.value = selectedUri.fsPath;
-			}
-		}, true));
-
-		this._register(this._imageRemoteRadioButton.onClicked(e => {
-			this._imageBrowseButton.style.display = 'none';
-			this._imageUrlLabel.innerText = this.urlPlaceholder;
-			this._imageUrlInputBox.setPlaceHolder(this.urlPlaceholder);
-		}));
-		this._register(this._imageLocalRadioButton.onClicked(e => {
-			this._imageBrowseButton.style.display = 'block';
-			this._imageUrlLabel.innerText = this.pathPlaceholder;
-			this._imageUrlInputBox.setPlaceHolder(this.pathPlaceholder);
-		}));
-		DOM.append(pathRow, inputContainer);
-
-		let embedRow = DOM.$('.row');
-		DOM.append(imageContentColumn, embedRow);
-		this._imageEmbedLabel = DOM.append(embedRow, DOM.$('.checkbox'));
-		this._imageEmbedCheckbox = new Checkbox(
-			this._imageEmbedLabel,
-			{
-				label: this.embedImageLabel,
-				checked: false,
-				onChange: (viaKeyboard) => { },
-				ariaLabel: this.embedImageLabel
-			});
-		DOM.append(embedRow, this._imageEmbedLabel);
-	}
-
-	private buildInsertLinkCallout(container: HTMLElement): void {
-		let linkContentColumn = DOM.$('.column.insert-link');
-		DOM.append(container, linkContentColumn);
-
-		let linkTextRow = DOM.$('.row');
-		DOM.append(linkContentColumn, linkTextRow);
-
-		this._linkTextLabel = DOM.$('p');
-		this._linkTextLabel.innerText = this.linkTextLabel;
-		DOM.append(linkTextRow, this._linkTextLabel);
-
-		const linkTextInputContainer = DOM.$('.input-field');
-		this._linkTextInputBox = new InputBox(
-			linkTextInputContainer,
-			this._contextViewService,
-			{
-				placeholder: this.linkTextPlaceholder,
-				ariaLabel: this.linkTextLabel
-			});
-		DOM.append(linkTextRow, linkTextInputContainer);
-
-		let linkAddressRow = DOM.$('.row');
-		DOM.append(linkContentColumn, linkAddressRow);
-		this._linkAddressLabel = DOM.$('p');
-		this._linkAddressLabel.innerText = this.linkAddressLabel;
-		DOM.append(linkAddressRow, this._linkAddressLabel);
-
-		const linkAddressInputContainer = DOM.$('.input-field');
-		this._linkUrlInputBox = new InputBox(
-			linkAddressInputContainer,
-			this._contextViewService,
-			{
-				placeholder: this.linkAddressPlaceholder,
-				ariaLabel: this.linkAddressLabel
-			});
-		DOM.append(linkAddressRow, linkAddressInputContainer);
-	}
-
-	private registerListeners(): void {
-		// Theme styler
-		if (this._calloutType === 'IMAGE') {
-			this._register(styler.attachInputBoxStyler(this._imageUrlInputBox, this._themeService));
-			this._register(styler.attachCheckboxStyler(this._imageEmbedCheckbox, this._themeService));
-		}
-		if (this._calloutType === 'LINK') {
-			this._register(styler.attachInputBoxStyler(this._linkTextInputBox, this._themeService));
-			this._register(styler.attachInputBoxStyler(this._linkUrlInputBox, this._themeService));
-		}
 	}
 
 	protected layout(height?: number): void {
 	}
 
-	public insert() {
+	public insert(): void {
```

Comment:
Functions like `insert` aren't something that's generic enough. It should be up to the superclasses to define what the custom buttons do.